### PR TITLE
Fix `test_fim/test_files/test_report_changes/test_disk_quota_values.py`

### LINKED ISSUE #1582: Fix `test_files/test_report_changes/test_disk_quota_values.py` in 4.2
## Description

| Related issue |
|--|
| #1516  #1539 |

|Development branch|
|:--:|
|1582-disk-quota-fix|

When it comes to starting with the test fixture, there is some information that can help you:
```
Module: File Integrity Monitoring
Target: Manager and Agent   
OS: Linux
```
I add some information obtained while researched the logs that failed.

|  Type | Description  |
|--|--|
|Test Path | test_fim/test_files/test_report_changes/test_disk_quota_values.py
| Consistent | Yes
| Test Execution | 3/3 executions failed
| Cases Failed |   [get_configuration25-tags_to_apply0] [get_configuration29-tags_to_apply0]
| Summary | `test_disk_quota_values.py` does not work properly for values of 7MB or higher 
| Error message |  "TimeoutError: Did not receive expected "The maximum configured size for the ... folder has been reached, ..." event."

In order to finish this issue the following tasks should be fulfilled:
- [x] Fails research.
   We have found out that test assumes that the folder being monitored has a size greater than the size limit entered. We need that the test creates a file with a bigger size.
- [x] Fix the test: The test needs to work for values of 7MB or higher.
- [x] Full Green/Yellow in `test_files/test_report_changes/test_disk_quota_values.py`. 
- [x] Documentation of any important change done for these tests or the used tools.

### PR DESCRIPTION
**Pull request by juliamagan**
|Related issue|
|---|
|Closes #1582|

### Environment

|OS|
|:--:|
|CentOS 8, Windows, Solaris, MacOS|
### Packages details

Type | Format | Architecture | Version | Revision | Tag | File name
-- | -- | -- | -- | -- | -- | --
Server | rpm | x86_64 | 4.2.0 | 40207 | v4.2.0-rc8 | wazuh-manager-4.2.0-1.1493.x86_64.rpm
Agent | rpm | x86_64 | 4.2.0 | 40207 | v4.2.0-rc8 | wazuh-agent-4.2.0-1.1493.x86_64.rpm
Agent | msi (Windows) | x86_64 | 4.2.0 | 40207 | v4.2.0-rc8 | wazuh-agent-4.2.0-1.1493.msi 
Agent | pkg (macOS) | x86_64 | 4.2.0 | 40207 | v4.2.0-rc8 | wazuh-agent-4.2.0-1.1493.pkg 
Agent | p5p (Solaris 11) | x86_64 | 4.2.0 | 40207 | v4.2.0-rc8 | wazuh-agent_v4.2.0-1.1493-sol11-i386.p5p​

#### Manager
```
syscheck.debug=2
analysisd.debug=2
monitord.rotate_log=0
```

#### Agent
```
agent.debug=2
syscheck.debug=2
monitord.rotate_log=0
```

### pytest_args
`-v --tier 0 --tier 1 --tier 2 --fim_mode="realtime" --fim_mode="whodata"`

## Description

After research #1493, we could see that some `FIM` tests were failing. In issue #1539 we specified which tests were failing and why. Then we created different issues for each error. 
This test was failing because was assuming that the size of the tested folder was greater than the limit size. We needed to change the tested folder and then, create a file with a size greater than the limit size.

## Tests results

||CentOS 8 Server|CentOS 8 Agent|Windows Agent|Solaris Agent|MacOS Agent|Reports|
|:--:|:--:|:--:|:--:|:--:|:--:|:--:|
|R1| 🟢 | 🟢  | 🟢  | 🟢 | 🟢  | [Tests results](https://github.com/wazuh/wazuh-qa/files/6860802/test_execution_1.zip)|
|R2| :green_circle: | 🟢 | 🟢  | 🟢  | 🟢  |[Tests results](https://github.com/wazuh/wazuh-qa/files/6860804/test_execution_2.zip)|
|R3| :green_circle: | 🟢 | 🟢 | 🟢  | 🟢  |[Tests results](https://github.com/wazuh/wazuh-qa/files/6860812/test_execution_3.zip)|

## Tests

- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.

### CODE REVIEW

**Comment by juliamagan** [comment]
While testing changes, we found out that macOS and Solaris are skipping this test because they don't support `realtime` or `whodata`. 
Since we have changed how parameters are generated, we need to add a specific case where macOS and Solaris use a low-frequency option instead of `realtime`.

**Review by Molter73 — CHANGES_REQUESTED** [review]
The PR is looking good! There are a few minor style changes I would like to see implemented.

I also ran the test on Ubuntu 20.04, Centos 8 and Windows, all of them with 2 cores and 2GB of ram, all tests passed without any issue:

#### Ubuntu
- [R0](https://github.com/wazuh/wazuh-qa/files/6867485/disk_quota-0.zip) 🟢
- [R1](https://github.com/wazuh/wazuh-qa/files/6867486/disk_quota-1.zip) 🟢
- [R2](https://github.com/wazuh/wazuh-qa/files/6867487/disk_quota-2.zip) 🟢

#### Centos
- [R0](https://github.com/wazuh/wazuh-qa/files/6867503/disk_quota-0.zip) 🟢
- [R1](https://github.com/wazuh/wazuh-qa/files/6867504/disk_quota-1.zip) 🟢
- [R2](https://github.com/wazuh/wazuh-qa/files/6867505/disk_quota-2.zip) 🟢

#### Windows
- [R0](https://github.com/wazuh/wazuh-qa/files/6867510/disk-quota-0.zip) 🟢
- [R1](https://github.com/wazuh/wazuh-qa/files/6867511/disk-quota-1.zip) 🟢
- [R2](https://github.com/wazuh/wazuh-qa/files/6867512/disk-quota-2.zip) 🟢

File: tests/integration/test_fim/test_files/test_report_changes/data/wazuh_disk_quota_values_conf.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        - file_size:
+            elements:
+            - enabled:
+                value: 'no'
+        - disk_quota:
+            elements:
+            - enabled:
+                value: 'yes'
+            - limit:
+                value: DISK_QUOTA_LIMIT
```
[the remark this answers is not in the document]
Comment (juliamagan) [reply tests/integration/test_fim/test_files/test_report_changes/data/wazuh_disk_quota_values_conf.yaml]: Done in 0d18840

File: tests/integration/test_fim/test_files/test_report_changes/test_disk_quota_values.py [hunk 2]
```diff
@@ -3,88 +3,69 @@
...
+    write_file(os.path.join(temp_dir, 'test'), random_string(translated_size*compression_ratio))
 
+    yield
 
-def extra_configuration_after_yield():
-    """
-    Restore syscheck.file_max_size internal option
-    """
-    restore_file_max_size()
+    remove_file(os.path.join(temp_dir, 'test'))
```
[the remark this answers is not in the document]
Comment (juliamagan) [reply tests/integration/test_fim/test_files/test_report_changes/test_disk_quota_values.py]: Done in bdada00

**Review by Molter73 — APPROVED** [review]
LGTM!

The changes from my previous review to this one are merely cosmetic so the tests ran previously are still valid IMO.

**Pull request merged**